c++/src/Reader.cc

```diff
@@ -526,16 +526,22 @@ namespace orc {
     if (metadata.get() == nullptr) {
       throw std::logic_error("No stripe statistics in file");
     }
+    auto currentStripeInfo = footer->stripes(static_cast<int>(stripeIndex));
```

Comment:
Should we consider some older compliers which don't support c++11 very well? I don't see anywhere else is using keyword auto.